### PR TITLE
Component/contact info

### PR DESCRIPTION
**Pull request by Rahaf-96**
### Content
- ContactInfo Component
    - Creating a form for user contact information using TextFields (name, phone, email, and postal code).
  
- Changes on TabsCheckout Component : 
   - Removing the App bar and CssBaseline from it.
   - Using the ContactInfo Component as the content of the first tab.
   - Validate the form inputs using the Yup Validation library.
   - Check the validation of the form using Formik.
   - Change the width of the checkout tabs layout, and remove the styling for the App bar.
---
### Screenshots

Screenshot from 2020-07-10 19-02-10
Screenshot from 2020-07-10 19-03-34
Screenshot from 2020-07-10 19-04-43

---
### Testing
on terminal use the following commands:
- pull origin component/contact-info
- npm run client
- go to http://localhost:3000/reservation on your browser.

### CODE REVIEW

**Comment by okjake** [comment]
Looks great so far! How are you going to track the form state? And when will the validation happen?

**Comment by Rahaf-96** [comment]
> Looks great so far! How are you going to track the form state? And when will the validation happen?

@okjake Actually when we searched how we can implement the validation of a multi-step form on the client side, we found that a library called Formik can give us this advantage. 
So we used Yup library for validation, and insert it into the checkout component using Formik.
To make things clear we followed the instructions mentioned in this article: https://medium.com/@nphivu414/build-a-multi-step-form-with-react-hooks-formik-yup-and-materialui-fa4f73545598

File: client/src/components/commonComponents/TabsCheckout/FormModel/checkoutFormModel.js [hunk 1]
```diff
@@ -0,0 +1,25 @@
+export default {
+  formId: 'checkoutForm',
+  formField: {
+    fullName: {
+      name: 'fullName',
+      label: 'full Name',
```
Comment (okjake) [inline client/src/components/commonComponents/TabsCheckout/FormModel/checkoutFormModel.js]: Pay attention to the capitalization in the labels and errors here, the form isn't totally consistent, I'd change
'full Name' -> 'Full Name',
'email' -> 'Email',
zip/Postal Code -> Zip/Postal Code

File: client/src/components/commonComponents/TabsCheckout/FormModel/validationSchema.js [hunk 2]
```diff
@@ -0,0 +1,21 @@
+import * as Yup from 'yup';
+import checkoutFormModel from './checkoutFormModel';
+const {
+  formField: { fullName, phone, email, zipCode },
+} = checkoutFormModel;
+const phoneRegExp = /^((\\+[1-9]{1,4}[ \\-]*)|(\\([0-9]{2,3}\\)[ \\-]*)|([0-9]{2,4})[ \\-]*)*?[0-9]{3,4}?[ \\-]*[0-9]{3,4}?$/;
+
+export default [
+  Yup.object().shape({
```
Comment (okjake) [inline client/src/components/commonComponents/TabsCheckout/FormModel/validationSchema.js:10]: Maybe you can share these schemas with the backend to apply the same validations there

**Review by okjake — APPROVED** [review]
Really nice work

**Review by sara219 — APPROVED** [review]

File: client/src/components/commonComponents/TabsCheckout/FormModel/checkoutFormModel.js (hunk 1 above)
Comment (Rahaf-96) [reply client/src/components/commonComponents/TabsCheckout/FormModel/checkoutFormModel.js]: Fixed :+1:

File: client/src/components/commonComponents/TabsCheckout/FormModel/validationSchema.js (hunk 2 above)
Comment (Rahaf-96) [reply client/src/components/commonComponents/TabsCheckout/FormModel/validationSchema.js:10]: yeah, we will do that.

**Review by tasneembhiri — APPROVED** [review]
:heart: :yellow_heart:

**Review by Jood80 — APPROVED** [review]

**Pull request merged**